examples/ensemble/plot_hgbt_regression.py

```diff
@@ -0,0 +1,397 @@
+- :ref:`monotonic_cst_gbdt`.
+- :ref:`interaction_cst_hgbt`.
+
+This example aims at showcasing points 2-5 in a real life setting.
+"""
+
+# %%
+# Author: Arturo Amor <david-arturo.amor-quiroz@inria.fr>
+#
+# License: BSD 3 clause
+#
+# Preparing the data
+# ==================
+# The `electricity dataset <http://www.openml.org/d/151>`_ consists of data
+# collected from the Australian New South Wales Electricity Market. In this
+# market, prices are not fixed and are affected by supply and demand. They are
+# set every five minutes. Electricity transfers to/from the neighboring state of
+# Victoria were done to alleviate fluctuations.
+#
+# The dataset (originally named ELEC2) contains 45,312 instances dated from 7
+# May 1996 to 5 December 1998. Each example of the dataset refers to a period of
+# 30 minutes, i.e. there are 48 instances for each time period of one day. Each
+# example on the dataset has 5 fields, the day of week, the time stamp, the New
+# South Wales electricity demand, the Victoria electricity demand. It is
+# originally a classification task, but here we use it as a regression where the
+# target is the scheduled electricity transfer between states.
+
+from sklearn.datasets import fetch_openml
+
+electricity = fetch_openml(
+    name="electricity", version=1, as_frame=True, parser="pandas"
+)
+df = electricity.frame
+X = df.drop(columns=["transfer", "class"])
+y = df["transfer"]
+X
+
+# %%
+# Let us explore the hourly electricity transfer over different days of the week:
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+colors = sns.color_palette("colorblind")
+
+fig, ax = plt.subplots(figsize=(15, 10))
+pointplot = sns.pointplot(x=df["period"], y=df["transfer"], hue=df["day"], ax=ax)
+handles, lables = ax.get_legend_handles_labels()
+ax.set(
+    title="Hourly energy transfer for different days of the week",
+    xticks=[i * 2 for i in range(24)],
+    xticklabels=list(range(24)),
+    xlabel="Time of the day",
+    ylabel="Normalized energy transfer",
+)
+_ = ax.legend(handles, ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"])
+
+# %%
+# Notice energy transfer increases systematically during weekends.
+#
+# Effect of number of trees and early stopping
+# ============================================
+# For the sake of illustrating the effect of the (maximum) number of trees, we
+# train a :class:`~sklearn.ensemble.HistGradientBoostingRegressor` over the
+# daily electricity transfer using the whole dataset. Then we visualize its
+# predictions depending on the `max_iter` parameter.
+
+from sklearn.ensemble import HistGradientBoostingRegressor
+
+max_iter_list = [10, 50]
+
+fig, ax = plt.subplots(figsize=(10, 5))
+average_week_demand = df.groupby(["day", "period"], observed=False)["transfer"].mean()
+average_week_demand.plot(color=colors[0], label="training data", linewidth=2, ax=ax)
+
+for idx, max_iter in enumerate(max_iter_list):
+    hgbt = HistGradientBoostingRegressor(max_iter=max_iter)
+    hgbt.fit(X, y)
```

Comment:
Would it work to do the same but a single split. I assume that we should see something similar when it comes to a separate test set.